worker/worker-execution/score-worker-execution-impl/src/main/java/io/cloudslang/worker/execution/services/ExecutionServiceImpl.java

```diff
@@ -226,6 +236,20 @@ private void updateMiIfRequired(Execution execution) {
         }
     }
 
+    private void updateUserInterrupts(Execution execution) {
```

Comment:
readExecutionObjectInterrupts() has been added to DebuggerBreakpointsHandlerImpl in the designer service. I think this code can be moved there. The interrupts for multiple branch executions are done in the debugger handler at line 119 and this can be done at the same time.

You can use DebuggerBreakpointsHandlerImpl::readExecutionObject to initialize your detached object